### PR TITLE
Add access statement

### LINKED ISSUE #4611: document: no new features except those that increase access
At our most recent [contributors conference](https://p5js.org/community/contributors-conference-2019.html), we made a commitment to not add any new features *except* those that increase access (meaning, inclusion and/or accessibility). This was intended not to shut down contributions, but to open up more conversations about what access means, and how we can continue to prioritize it. It would be great to get a write up / explanation of this in this repo so we can understand and apply it more easily. I think @outofambit wrote some text around this already we could start with. The doc could live in the `contributor_docs` section.

### PR DESCRIPTION
**Pull request by outofambit**
closes #4611 

here's my draft of this statement! i took some of my original notes from the contributors conference, added in some language from #4611 itself, and expanded it some too capture a little more context. any + all feedback welcome! 🙏🏼

i added a link to this statement from `contributor_docs/README.md`, are there other places we should link to it from?

### CODE REVIEW

File: contributor_docs/access.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Our Focus on Access
+
+At the [2019 Contributors Conference](https://p5js.org/community/contributors-conference-2019.html), we made a commitment to only add features to p5.js that increase access (meaning inclusion and/or accessibility). This means we prioritize the needs of marginalized groups over the continued comfort of more privileged groups with p5.js.
+
+This is not the first conversation we've had about access and inclusion. We share similar intentions in our [Community Statement](../CODE_OF_CONDUCT.md), which was written at the [2015 Contributors Conference](https://p5js.org/community/contributors-conference-2015.html).
```
Comment (lmccart) [inline contributor_docs/access.md:5]: Maybe reword "This is not the first conversation we've had about access and inclusion..." to "This is part of an ongoing conversation about access and inclusion within p5.js. Our intention to hold these as core values from which p5.js is built is laid out in our [Community Statement]..."

File: contributor_docs/access.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+**Please consider this a starting point.** We want to invite more conversations about what access means and how we can prioritize it.
+
+## Kinds of access
+
+Increasing access is not a focus on expanding the raw number of people in the p5.js community. It is a focus on making p5.js available to and approachable for more people who are excluded from the p5.js community (intentionally or not) and people who are excluded from similar tools.
```
Comment (lmccart) [inline contributor_docs/access.md:11]: for clarity of language and conciseness, suggestion to change "approachable for more people" to "approachable for people"

File: contributor_docs/access.md (hunk 1 above)
Comment (lmccart) [reply contributor_docs/access.md:5]: Maybe one more statement to the effect of: "We engage in accessibility work in the context of anti-racist work, and believe it's necessary to understand the ways that factors such as disability, gender, sexuality, and class intersect with, and disproportionately affect people based on race." I don't know if that's quite it but trying to get there...

File: contributor_docs/access.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+Access here means making p5.js better for:
+
+- People who speak Spanish, Chinese, or other languages besides English
```
Comment (lmccart) [inline contributor_docs/access.md:15]: add Korean (launching soon). or maybe remove specific languages altogether?

File: contributor_docs/access.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+- People who are Black, Indigenous, or people of Color
```
Comment (lmccart) [inline contributor_docs/access.md:16]: capitalize people or lc Color

File: contributor_docs/access.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+- People who are lesbian, gay, bisexual, trans, or queer
+- People with marginalized genders
+- People who are disabled or sick
```
Comment (lmccart) [inline contributor_docs/access.md:19]: + An item about income / class

File: contributor_docs/access.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+- and other people who are systemically excluded
+
+### Examples
+
+These are examples of efforts we believe to increase access:
+
+- Translating more documentation and other materials into more languages
+- Improving out support for assistive devices (such as screenreaders)
+- Making p5.js error messages more helpful and supportive to developers
```
Comment (lmccart) [inline contributor_docs/access.md:28]: + Teaching p5.js to groups that are marginalized or excluded within coding and/or the arts.

Comment (lmccart) [inline contributor_docs/access.md:28]: replace "developers" with "people using the tool"

File: contributor_docs/access.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+There are other things we haven't thought of yet and we're excited to figure out what they are together. If you have an idea, please [share it as an issue](https://github.com/processing/p5.js/issues/new/choose).
+
+## Maintenance
+
+We also affirm our intention to maintain the existing feature set of p5.js. We'd like to fix bugs regardless of which area of the codebase they're in or how closely they are related to our focus on access.
```
Comment (lmccart) [inline contributor_docs/access.md:34]: maybe replace "or how closely they are related to our focus on access." with something like "because we believe consistency of the tool makes it more accessible for beginners."

**Comment by lmccart** [comment]
@dorothysantos would love your eyes on this, also johanna hedva who I will email

File: contributor_docs/access.md (hunk 3 above)
Comment (montoyamoraga) [reply contributor_docs/access.md:15]: i vote removing specific languages altogether so it's easy to maintain and there are no hierarchies :)

File: contributor_docs/access.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+- People with marginalized genders
+- People who are disabled or sick
+- and other people who are systemically excluded
+
+### Examples
+
+These are examples of efforts we believe to increase access:
+
+- Translating more documentation and other materials into more languages
+- Improving out support for assistive devices (such as screenreaders)
```
Comment (limzykenneth) [inline contributor_docs/access.md:27]:
```suggestion
- Improving our support for assistive devices (such as screenreaders)
```

Typo

File: contributor_docs/README.md [hunk 9]
```diff
@@ -8,6 +8,12 @@ This project follows the [all-contributors](https://github.com/kentcdodds/all-co
 
 
 
+# Prioritizing access
+
+We are prioritizing work that expands access to p5.js! See [our access statement](./access.md) for more details.
```
Comment (lm-n) [inline contributor_docs/README.md:13]:
```suggestion
We are prioritizing work that expands access (inclusion and accessibility) to p5.js! See [our access statement](./access.md) for more details.
```

File: contributor_docs/access.md (hunk 2 above)
Comment (lm-n) [inline contributor_docs/access.md:11]:
```suggestion
Increasing access is not a focus on expanding the raw number of people in the p5.js community. It is a focus on making p5.js available to and approachable for people who are excluded from the p5.js community (intentionally or not) and from similar tools and communities.
```

File: contributor_docs/access.md (hunk 3 above)
Comment (lm-n) [inline contributor_docs/access.md:15]:
```suggestion
- People who speak languages other than English
```

File: contributor_docs/access.md (hunk 4 above)
Comment (lm-n) [inline contributor_docs/access.md:16]:
```suggestion
- Black people, Indigenous peoples, and people of Color
```
changing wording and "or" for "and"

File: contributor_docs/access.md (hunk 5 above)
Comment (lm-n) [reply contributor_docs/access.md:19]: - also add item about rural communities

File: contributor_docs/access.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+Increasing access is not a focus on expanding the raw number of people in the p5.js community. It is a focus on making p5.js available to and approachable for more people who are excluded from the p5.js community (intentionally or not) and people who are excluded from similar tools.
+
+Access here means making p5.js better for:
+
+- People who speak Spanish, Chinese, or other languages besides English
+- People who are Black, Indigenous, or people of Color
+- People who are lesbian, gay, bisexual, trans, or queer
+- People with marginalized genders
+- People who are disabled or sick
+- and other people who are systemically excluded
```
Comment (lm-n) [inline contributor_docs/access.md:20]:
```suggestion
- and other people who are systemically excluded and traditionally underrepresented
```

File: contributor_docs/access.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
+# Our Focus on Access
+
+At the [2019 Contributors Conference](https://p5js.org/community/contributors-conference-2019.html), we made a commitment to only add features to p5.js that increase access (meaning inclusion and/or accessibility). This means we prioritize the needs of marginalized groups over the continued comfort of more privileged groups with p5.js.
```
Comment (lm-n) [inline contributor_docs/access.md:3]: I'm not sure about the use of marginalized and whether it reinforces marginalization. Maybe say something like historically excluded and underrepresented?

Comment (lm-n) [inline contributor_docs/access.md:3]:
```suggestion
At the [2019 Contributors Conference](https://p5js.org/community/contributors-conference-2019.html), we made a commitment to only add features to p5.js that increase access (meaning inclusion and/or accessibility). This means considering the vectors of diversity (e.g. gender, social, economic, race, ethnicity, language, ability, etc.) that can impact access/participation, and taking action to acknowledge, dismantle, and prevent barriers. We prioritize the needs of historically marginalized groups over the continued comfort of more privileged groups with p5.js.
```

File: contributor_docs/access.md (hunk 5 above)
Comment (lm-n) [inline contributor_docs/access.md:19]:
```suggestion
- People with disabilities or illness
```

File: contributor_docs/access.md (hunk 8 above)
Comment (lm-n) [inline contributor_docs/access.md:27]:
```suggestion
- Improving our support for assistive technologies (such as screenreaders)
- Following Web Content Accessibility Guidelines in our tools and working towards making it easier for users to follow them in their projects.
```

File: contributor_docs/access.md (hunk 6 above)
Comment (lm-n) [reply contributor_docs/access.md:28]: This is really important but I think we could write in a different way... "Teaching p5.js to people that are marginalized or excluded..." could be read as if we were on missionary quest to evangelize people. 
Maybe: 
- Supporting learning and teaching p5.js in communities that have been historically marginalized, excluded, and underrepresented in creative coding and digital arts???  

Just an idea, I think it still needs improvement. @montoyamoraga @lmccart @outofambit  what do you think/help?

File: contributor_docs/access.md (hunk 11 above)
Comment (outofambit) [reply contributor_docs/access.md:3]: @lm-n i love this, thank you!

File: contributor_docs/access.md (hunk 1 above)
Comment (outofambit) [reply contributor_docs/access.md:5]: > This is part of an ongoing conversation

@lmccart i like this framing a lot better thank you! will think about phrasing for the "we engage" statement. if anyone else has ideas for that please share! :)

File: contributor_docs/access.md (hunk 10 above)
Comment (outofambit) [reply contributor_docs/access.md:20]: @lm-n would you be ok with dropping the word "traditionally" here? or perhaps we could use "historically" instead? i don't feel strongly about this, just want to ask! feel free to say no.
```suggestion
- and other people who are systemically excluded and historically underrepresented
```

File: contributor_docs/access.md (hunk 2 above)
Comment (outofambit) [reply contributor_docs/access.md]: changed this with @lm-n's suggestion!

File: contributor_docs/access.md (hunk 3 above)
Comment (outofambit) [reply contributor_docs/access.md]: done!

File: contributor_docs/access.md (hunk 4 above)
Comment (outofambit) [reply contributor_docs/access.md]: ✅

File: contributor_docs/access.md [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+Increasing access is not a focus on expanding the raw number of people in the p5.js community. It is a focus on making p5.js available to and approachable for people who are excluded from the p5.js community (intentionally or not) and from similar tools and communities.
+
+Access here means making p5.js better for:
+
+- People who speak languages other than English
+- Black people, Indigenous peoples, and People of Color
+- People who are lesbian, gay, bisexual, trans, or queer
+- People with marginalized genders
+- People with disabilities or illness
+- and other people who are systemically excluded and historically underrepresented
```
Comment (lm-n) [inline contributor_docs/access.md:20]: I'm also wondering if we should add novices/beginners here. 
```suggestion
- People with little or no prior experience in open source and creative coding 
- and other people who are systemically excluded and historically underrepresented
```

File: contributor_docs/access.md (hunk 6 above)
Comment (outofambit) [reply contributor_docs/access.md]: @lm-n that's a good point! i like @lmccart's because it highlights the act of teaching itself (we have other items that cover documentation and coding work already). hmm, how about something like this?

* Mentoring and supporting learners of p5.js _within_ communities that are historically excluded from and marginalized in creative coding and the digital arts

Comment (outofambit) [reply contributor_docs/access.md]: ✅

File: contributor_docs/access.md (hunk 12 above)
Comment (outofambit) [reply contributor_docs/access.md:20]: yes! i had a thought about that before but forgot to write something!

File: contributor_docs/access.md (hunk 6 above)
Comment (lm-n) [reply contributor_docs/access.md]: I like "Mentoring and supporting learners of p5.js within communities that are historically excluded from and marginalized in creative coding and the digital arts" 😸 🥰

File: contributor_docs/access.md (hunk 5 above)
Comment (lmccart) [reply contributor_docs/access.md]: For the first point above, maybe:
- People who lack opportunities and/or resources to engage with creative coding due to class or income

**Comment by limzykenneth** [comment]
Having a quick read over it, is there somewhere that mentions not accepting new feature requests except those that increase access? I may have just missed it or it is documented somewhere else?

Looks awesome btw!

**Comment by lmccart** [comment]
@limzykenneth good point. I think we had talked somewhere about the idea of also putting a question about that in the new feature issue template. so we might want to mention that flow here.

**Comment by limzykenneth** [comment]
@lmccart I think @montoyamoraga was talking about that, but would definitely be nice to document it clearly here too.

File: contributor_docs/access.md [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+Increasing access is not a focus on expanding the raw number of people in the p5.js community. It is a focus on making p5.js available to and approachable for people who are excluded from the p5.js community (intentionally or not) and from similar tools and communities.
+
+Access here means making p5.js better for:
+
+- People who speak languages other than English
+- Black people, Indigenous peoples, and People of Color
+- People who are lesbian, gay, bisexual, trans, or queer
+- People with marginalized genders
+- People with disabilities or illness
+- People with little or no prior experience in open source and creative coding
```
Comment (lmccart) [inline contributor_docs/access.md:20]: my one nit-picky question here is how do we determine "little prior experience"? would it include anyone that is a student? would it include someone that has mastered the basics of coding? i'm trying to get specific here because i see it would be easy to make an argument for most features being helpful for to those with little experience, given that is our majority user demographic.

File: contributor_docs/access.md (hunk 5 above)
Comment (montoyamoraga) [reply contributor_docs/access.md]: @lm-n what about
People from the Global South
?

File: contributor_docs/access.md (hunk 13 above)
Comment (montoyamoraga) [reply contributor_docs/access.md:20]: i agree, it's hard to draw the line of what features should be added, i think that it is hard to grasp all the use cases of the p5.js features, so people can easily propose a new feature, and adding it seems like little work, but it involves a lot of work on the development, or the proposed feature, is a corner case of new features.

something that i am always thinking is that new features, longer documentation, more dependencies add more chances for breaking, adds more barriers to people who want to read the source code, or help out. i will keep on thinking about this too, i don't know a good way to phrase this or explain this :)

Comment (drsantos8791) [reply contributor_docs/access.md:20]: Great start! But I want us to start expanding and going beyond words such as "'underrepresented’, ‘underserved’, ‘marginalized’ or ‘minority." I know this is a longer conversation about semantics, but in the past few years, there has been a push to not use these words. More soon!

File: contributor_docs/access.md [hunk 14]
```diff
@@ -0,0 +1,40 @@
+# Our Focus on Access
+
+At the [2019 Contributors Conference](https://p5js.org/community/contributors-conference-2019.html), we made a commitment to only add features to p5.js that increase access (meaning inclusion and/or accessibility). This means considering the vectors of diversity (e.g. gender, social, economic, race, ethnicity, language, ability, etc.) that can impact access/participation; and taking action to acknowledge, dismantle, and prevent barriers. We prioritize the needs of historically marginalized groups over the continued comfort of more privileged groups with p5.js.
+
+We will not accept feature requests that don't support our effort to increase access. You'll see this criteria reflected in our issue and pull request templates.
```
Comment (outofambit) [inline contributor_docs/access.md:5]: @limzykenneth @lmccart @montoyamoraga this seem okay for being explicit about not accepting some work based on this criteria?

File: contributor_docs/access.md (hunk 1 above)
Comment (outofambit) [reply contributor_docs/access.md]: @lmccart does this line we added cover this point you had? or is there another point to make sure we cover?

> This means considering the vectors of diversity (e.g. gender, social, economic, race, ethnicity, language, ability, etc.) that can impact access/participation; and taking action to acknowledge, dismantle, and prevent barriers.

File: contributor_docs/access.md (hunk 5 above)
Comment (outofambit) [reply contributor_docs/access.md]: i added the class/income language from @lmccart :)

File: contributor_docs/access.md (hunk 6 above)
Comment (outofambit) [reply contributor_docs/access.md]: added!

File: contributor_docs/access.md (hunk 5 above)
Comment (lm-n) [reply contributor_docs/access.md]: @montoyamoraga I feel uncomfortable with Global South... @dorothysantos do you have any ideas?

File: contributor_docs/access.md (hunk 14 above)
Comment (limzykenneth) [reply contributor_docs/access.md:5]: Seems good to me.

File: contributor_docs/access.md [hunk 15]
```diff
@@ -0,0 +1,40 @@
+# Our Focus on Access
+
+At the [2019 Contributors Conference](https://p5js.org/community/contributors-conference-2019.html), we made a commitment to only add features to p5.js that increase access (meaning inclusion and/or accessibility). This means considering the vectors of diversity (e.g. gender, social, economic, race, ethnicity, language, ability, etc.) that can impact access/participation; and taking action to acknowledge, dismantle, and prevent barriers. We prioritize the needs of historically marginalized groups over the continued comfort of more privileged groups with p5.js.
```
Comment (agenusofmoth) [inline contributor_docs/access.md:3]: please change "ability" to "disability." this is preferred terminology when speaking about the disability community, as it foregrounds the "dis," which is the political part.

File: contributor_docs/access.md (hunk 13 above)
Comment (lmccart) [reply contributor_docs/access.md]: @drsantos8791 I think we may need more help here. Do you have suggestions for alternative phrasing? For example in these phrases: 
- `We prioritize the needs of historically marginalized groups over the continued comfort of more privileged groups with p5.js.`
- `Mentoring and supporting learners of p5.js within communities that are historically excluded from and marginalized in creative coding and the digital arts`

File: contributor_docs/access.md (hunk 15 above)
Comment (outofambit) [reply contributor_docs/access.md]: thank you @agenusofmoth! made this change in ce00df9.

File: contributor_docs/access.md (hunk 14 above)
Comment (lm-n) [reply contributor_docs/access.md:5]: 👏🏼

File: contributor_docs/access.md (hunk 5 above)
Comment (lm-n) [reply contributor_docs/access.md]: @drsantos8791 just realized I tagged your other github account. Can you help us with the people that are not from North America/Europe?

File: contributor_docs/access.md (hunk 13 above)
Comment (drsantos8791) [reply contributor_docs/access.md]: @lmccart At the moment, and after reading through several articles on this issue, I don't have any further suggestions. Also, I feel these statements best exemplify the mission and objective of what we're trying to do. If I come across suggestions, I will gladly share them with the group. But at the moment, I don't have alternatives. Thank you for checking in.

Comment (lmccart) [reply contributor_docs/access.md]: Thank you @drsantos8791! And this is a living document so we can continue to update as we learn more and develop our language and thinking.

File: contributor_docs/access.md (hunk 5 above)
Comment (drsantos8791) [reply contributor_docs/access.md]: @lm-n Not a problem! It was easier for me to create a PF-specific GitHub account. Pardon the confusion. 

As for the "Global South," I understand your discomfort because in places such as the Philippines, for instance, the country (and in the education and tech spheres/realms) experience "brain drain" (not the best terminology, but this phenomenon has been documented). Essentially, even within the Filipino community, class is a huge issue that can't be easily remedied by support from a US-based program because you may be providing this opportunity to someone who readily has access to these opportunities within their region or country. This topic is such a thorny one, to be honest. I have too many thoughts for a comment, honestly. But I can see how we might flatten experiences and this is why transnational perspectives are super important. @montoyamoraga @lmccart @xinemata We can discuss for the future! Definitely an important issue especially as we think about translation projects.

File: contributor_docs/access.md (hunk 1 above)
Comment (lmccart) [reply contributor_docs/access.md]: @outofambit I want to take a crack at one more line after that paragraph that speaks to intersectionality. I think right now we are mentioning the different vectors to consider within diversity but we might recognize how these play into each other and shouldn't be treated in isolation. I can work on this in the next couple days and propose something.

**Comment by drsantos8791** [comment]
Oh, I get these notifications. LOL. Sigh. I will pay closer attention.

Dorothy R. Santos
Program Manager, Processing Foundation
Pronouns I respond to: she/her/hers and they/them/theirs
dorothy@processing.org
+1 415 748 2679 tel

File: contributor_docs/access.md (hunk 5 above)
Comment (lmccart) [reply contributor_docs/access.md]: @lm-n @montoyamoraga @drsantos8791 this doesn't address all the issues here by any means, but could "non-western" be a useful term?

Comment (lm-n) [reply contributor_docs/access.md]: @lmccart I definitely think "non-western" is better than "Global South"! But in translation it might get lost, in Spanish language the idea of the "west" includes Latin America and the Caribbean while in American English these regions are "non-western."
